src/lis3dh-motion-detection.cpp

```diff
@@ -191,7 +191,64 @@ status_t LIS3DH::writeRegister(uint8_t offset, uint8_t dataToWrite) {
     returnError = IMU_HW_ERROR;
   }
 
+	_DEBBUG("Write register 0x", offset, " = ", dataToWrite);
 	return returnError;
+
+	delay(100);             // some users reported hangs without delay.
+}
+
+// Read tap (aka: 'click')
+uint8_t LIS3DH::readTap(){
+	uint8_t t;
+	
+	readRegister(&t, LIS3DH_CLICK_SRC); // read
+	return t;
+}
+
+uint8_t LIS3DH::readEvent(){
+	uint8_t t;
+	
+	readRegister(&t, LIS3DH_INT1_SRC);
+	return t;
+}
+
+void LIS3DH::configTap(bool Zdouble, bool Zsingle, bool Ydouble, bool Ysingle, bool Xdouble, bool Xsingle, uint8_t threshold){
+	uint8_t r = 0;
+
+	writeRegister(LIS3DH_CTRL_REG3, 0x80); 				// Tap on INT1
+	readRegister(&r, LIS3DH_CTRL_REG5);
+	writeRegister(LIS3DH_CTRL_REG5, r | 0b1000); 			// latch INT1
```

Comment:
Why not leave the user to determine if latch is required or not?

I believe we're over complicating the library, if a user needs latching and read the register and source of the interrupts, the user is probably experienced and therefore would not use this library.

Reply:
You have right, I thought it was a sane default. Well the library has many holes.p. I will add it

(I don’t understood the last argument, but no worries)